neqo-transport/src/qlog.rs

```diff
@@ -20,28 +22,40 @@ use qlog::events::{
     },
     EventData, RawInfo,
 };
+#[cfg(feature = "qlog")]
 use smallvec::SmallVec;
 
 use crate::{
     connection::State,
-    frame::{CloseError, Frame},
-    packet::{self, metadata::Direction},
+    packet,
     path::PathRef,
     recovery::sent,
+    tparams::TransportParametersHandler,
+    version::{self, Version},
+};
+#[cfg(feature = "qlog")]
+use crate::{
+    frame::{CloseError, Frame},
+    packet::metadata::Direction,
     stream_id::StreamType as NeqoStreamType,
-    tparams::{
-        TransportParameterId::{
-            self, AckDelayExponent, ActiveConnectionIdLimit, DisableMigration, InitialMaxData,
-            InitialMaxStreamDataBidiLocal, InitialMaxStreamDataBidiRemote, InitialMaxStreamDataUni,
-            InitialMaxStreamsBidi, InitialMaxStreamsUni, MaxAckDelay, MaxUdpPayloadSize,
-            OriginalDestinationConnectionId, StatelessResetToken,
-        },
-        TransportParametersHandler,
+    tparams::TransportParameterId::{
+        self, AckDelayExponent, ActiveConnectionIdLimit, DisableMigration, InitialMaxData,
+        InitialMaxStreamDataBidiLocal, InitialMaxStreamDataBidiRemote, InitialMaxStreamDataUni,
+        InitialMaxStreamsBidi, InitialMaxStreamsUni, MaxAckDelay, MaxUdpPayloadSize,
+        OriginalDestinationConnectionId, StatelessResetToken,
     },
-    version::{self, Version},
 };
 
+#[cfg_attr(
+    not(feature = "qlog"),
+    expect(
+        unused_variables,
+        clippy::missing_const_for_fn,
+        reason = "Only used with qlog."
+    )
+)]
 pub fn connection_tparams_set(qlog: &Qlog, tph: &TransportParametersHandler, now: Instant) {
+    #[cfg(feature = "qlog")]
     qlog.add_event_data_with_instant(
```

Comment:
If `add_event_data_with_instant` does nothing when qlog is disabled, do you really need all the guards around it?  The same applies for all these functions.

Reply:
I think so, because otherwise things from the `quiche/qlog` crate are used, which is not a dep.